salt/transport/client.py

```diff
@@ -91,86 +107,217 @@ def factory(opts, **kwargs):
         )
 
 
-# TODO: better doc strings
-class AsyncChannel:
+class AsyncReqChannel:
     """
-    Parent class for Async communication channels
+    Factory class to create a asynchronous communication channels to the
+    master's ReqServer. ReqChannels connect to the master's ReqServerChannel on
+    the minion's master_port (default: 4506) option.
     """
 
-    # Resolver is used by Tornado TCPClient.
-    # This static field is shared between
-    # AsyncReqChannel and AsyncPubChannel.
-    # This will check to make sure the Resolver
-    # is configured before first use.
-    _resolver_configured = False
-
-    @classmethod
-    def _config_resolver(cls, num_threads=10):
-        import salt.ext.tornado.netutil
-
-        salt.ext.tornado.netutil.Resolver.configure(
-            "salt.ext.tornado.netutil.ThreadedResolver", num_threads=num_threads
-        )
-        cls._resolver_configured = True
-
-
-# TODO: better doc strings
-class AsyncReqChannel(AsyncChannel):
-    """
-    Factory class to create a Async communication channels to the ReqServer
-    """
+    async_methods = [
+        "crypted_transfer_decode_dictentry",
+        "_crypted_transfer",
+        "_uncrypted_transfer",
+        "send",
+    ]
+    close_methods = [
+        "close",
+    ]
 
     @classmethod
     def factory(cls, opts, **kwargs):
+        import salt.ext.tornado.ioloop
```

Comment:
Now that we import it at the top, do we still need this local import?

Reply:
Unfortunately, for now, yes. Because we are importing `salt.transport.zeromq` and `salt.transport.tcp` when the transport is chosen we can't reference any salt modules without importing them in the local scope of this method. Ideally, we can sort this out as the changes get further along. Right now my main focus is just shuffling stuff around to reduce code duplication and separate our 'business' logic from 'transport' logic.

Reply:
Ok.